FantasyUniverse/Views/QuestionView.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  QuestionView.swift
+//  FantasyUniverse
+//
+//  Created by TarasPeregrinus on 01.02.2022.
+//
+
+import SwiftUI
+
+struct QuestionView: View {
+    
+    @Binding var questions: [Question]
+    @State var currentQuestion: Question
+    @Binding var correct: Int
+    @Binding var wrong: Int
+    @Binding var answered: Int
+    @State var selected = ""
+    
+    var body: some View {
+        
+        VStack(spacing: 20) {
+            // Question
+            QuestionLabelView(text: currentQuestion.question!)
+                .lineLimit(4)
+            Spacer()
+            
+            OptionButton(color: buttonColor(option: currentQuestion.optionA!),
+                         text: currentQuestion.optionA!,
+                         action: {
+                selected = currentQuestion.optionA!
+            })
+                .disabled(selected.isEmpty ? false : true)
+                .opacity(selected.isEmpty ? 1 : 0.7)
+            
+            OptionButton(color: buttonColor(option: currentQuestion.optionB!),
+                         text: currentQuestion.optionB!,
+                         action: {
+                selected = currentQuestion.optionB!
+            }).lineLimit(nil)
+                .disabled(selected.isEmpty ? false : true)
+                .opacity(selected.isEmpty ? 1 : 0.7)
+            
+            OptionButton(color: buttonColor(option: currentQuestion.optionC!),
+                         text: currentQuestion.optionC!,
+                         action: {
+                selected = currentQuestion.optionC!
+            }).lineLimit(nil)
+                .disabled(selected.isEmpty ? false : true)
+                .opacity(selected.isEmpty ? 1 : 0.7)
+            
+            OptionButton(color: buttonColor(option: currentQuestion.optionD!),
+                         text: currentQuestion.optionD!,
+                         action: {
+                selected = currentQuestion.optionD!
+            }).lineLimit(nil)
+                .disabled(selected.isEmpty ? false : true)
+                .opacity(selected.isEmpty ? 1 : 0.7)
```

Comment:
Maybe wrapping these in a separate wrapper and disabling/styling it would help with the code repetition?